src/sign_workflow/signer.py

```diff
@@ -19,40 +18,40 @@
 
 
 class Signer:
-
     ACCEPTED_FILE_TYPES = [".zip", ".jar", ".war", ".pom", ".module", ".tar.gz"]
 
     def __init__(self):
         self.git_repo = GitRepository(self.get_repo_url(), "HEAD", working_subdirectory="src")
         self.git_repo.execute("./bootstrap")
         self.git_repo.execute("rm config.cfg")
 
-    def sign_artifacts(self, artifacts, basepath):
+    def sign_artifact(self, artifact, basepath, signature_type):
+        self.generate_signature_and_verify(artifact, basepath, signature_type)
+
+    def sign_artifacts(self, artifacts, basepath, signature_type):
         for artifact in artifacts:
             if not self.is_valid_file_type(artifact):
                 logging.info(f"Skipping signing of file ${artifact}")
                 continue
-            location = os.path.join(basepath, artifact)
-            self.sign(location)
-            self.verify(location + ".asc")
+            self.generate_signature_and_verify(artifact, basepath, signature_type)
+
+    def generate_signature_and_verify(self, artifact, basepath, signature_type):
+        location = os.path.join(basepath, artifact)
+        self.sign(location, signature_type)
+        self.verify(location + signature_type)
 
     def is_valid_file_type(self, file_name):
         return any(
-            x
-            in [
-                pathlib.Path(file_name).suffix,
-                "".join(pathlib.Path(file_name).suffixes),
-            ]
-            for x in Signer.ACCEPTED_FILE_TYPES
+            file_name.endswith(x) for x in Signer.ACCEPTED_FILE_TYPES
```

Comment:
Use `os.path.splitext` to get the extension from the file name, then `in` to check ether it’s one of the accepted types.

Reply:
That does not work since we have artifacts with names like `opensearch-1.0.0.tar.gz`. That is why I changed it to `endswith`

Reply:
We could just change .tar.gz into .gz. Technically the extension of that file is not .tar.gz, it’s .gz.